packages/maker/src/api/defineOntology.ts

```diff
@@ -67,63 +61,140 @@ export async function defineOntology(
-                  ),
+              interfaceType: convertInterface(
+                description,
+                displayName,
+                apiName,
+                properties,
               ),
             }];
           },
         ),
     ),
+    blockPermissionInformation: {
+      actionTypes: {},
+      linkTypes: {},
+      objectTypes: {},
+    },
+  };
+}
+
+function convertInterface(
+  description: string | undefined,
+  displayName: string,
+  apiName: string,
+  properties: Record<string, SharedPropertyType>,
+): OntologyIrInterfaceType {
+  return {
+    displayMetadata: {
+      description,
+      displayName: displayName ?? apiName,
+      icon: undefined,
+    },
+    apiName,
+    extendsInterfaces: [],
+    links: [],
+    status: {
+      active: true,
+      type: "active",
+    },
+    allExtendsInterfaces: [],
+    allLinks: [],
+    allProperties: [],
+    properties: Object.values(properties)
+      .map<OntologyIrSharedPropertyType>((spt) => convertSpt(spt)),
   };
 }
 
-export function dumpOntologyFullMetadata(): Gateway.OntologyFullMetadata {
+export function dumpOntologyFullMetadata(): OntologyIrOntologyBlockDataV2 {
   return convertToWireOntology(ontologyDefinition);
 }
 
 function convertSpt(
   { type, array, description, apiName, displayName }: SharedPropertyType,
-): Gateway.SharedPropertyType {
+): OntologyIrSharedPropertyType {
   return {
-    rid: "ri.ontology.main.generated-object.foo",
     apiName,
-    displayName: displayName ?? apiName,
-    description,
-    dataType: array
-      ? { type: "array" as const, subType: { type } }
-      : { type },
+    displayMetadata: {
+      displayName: displayName ?? apiName,
+      visibility: "NORMAL",
+      description,
+    },
+    type: array
+      ? {
+        type: "array" as const,
+        array: {
+          subtype: convertType(type),
+        },
+      }
+      : convertType(type),
+    aliases: [],
+    baseFormatter: undefined,
+    dataConstraints: undefined,
+    gothamMapping: undefined,
+    indexedForSearch: true,
+    provenance: undefined,
+    typeClasses: [],
+    valueType: undefined,
   };
 }
+
+function convertType(
+  type: PropertyTypeType,
+): Type {
+  switch (type) {
+    case "marking":
+      return { type, marking: { markingType: "MANDATORY" } };
+
+    case "geopoint":
+      return { type: "geohash", geohash: {} };
+
+    case "decimal":
+      return { type, [type]: { precision: undefined, scale: undefined } };
+
+    case "string":
+      return {
+        type,
+        [type]: {
+          analyzerOverride: undefined,
+          enableAsciiFolding: undefined,
+          isLongText: false,
+          supportsExactMatching: true,
+        },
+      };
+
+    default:
+      // use helper function to distribute `type` properly
+      return asdf(type);
+  }
+}
+
+/**
+ * Helper function to avoid duplication
+ * @param type
+ * @returns
+ */
+function asdf<T extends string>(
```

Comment:
can we rename this function please?

Reply:
😬 that's embarrassing. when i refactor i just throw up a ton of crap and worry about renaming later